### PR TITLE
Support person data in rollbar.js

### PR DESCRIPTION
**Pull request by and-megan**

### CODE REVIEW

**Comment by and-megan** [comment]
@jondeandres This works but it's automatic, ie you don't need to list it in the config.js_options, - should this be optional or is it ok as is?

**Comment by and-megan** [comment]
I made it optional, but I can take it out

File: lib/rollbar/middleware/js.rb [hunk 1]
```diff
@@ -103,7 +107,12 @@ def close_old_response(response)
       end
 
       def config_js_tag(env)
-        script_tag("var _rollbarConfig = #{config[:options].to_json};", env)
+        js_config = config[:options].dup
+        if js_config[:payload][:person_tracking]
```
Comment (rokob) [inline lib/rollbar/middleware/js.rb]: If `:payload` does not exist in the config then this throws

File: lib/rollbar/middleware/js.rb [hunk 2]
```diff
@@ -103,7 +107,12 @@ def close_old_response(response)
...
+          person_data = extract_person_data_from_controller(env)
+          js_config[:payload][:person] = person_data if person_data
```
Comment (rokob) [inline lib/rollbar/middleware/js.rb]: Same here

**Comment by and-megan** [comment]
@rokob I think it's fixed now

File: lib/rollbar/middleware/js.rb [hunk 3]
```diff
@@ -103,7 +107,12 @@ def close_old_response(response)
       end
 
       def config_js_tag(env)
-        script_tag("var _rollbarConfig = #{config[:options].to_json};", env)
+        js_config = config[:options].dup
+        if js_config.try(:[], :payload).try(:[], :person_tracking)
```
Comment (jondeandres) [inline lib/rollbar/middleware/js.rb]: I think we just need to do:

```ruby
person_data = extract_person_data_from_controller(env)
if person_data
  js_config[:payload] ||= {} # We initialize the `payload` dictionary just with `{}` if it doesn't exist
  js_config[:payload][:person] = person_data
end
```

**Comment by and-megan** [comment]
@jondeandres Looks good. I had :person_tracking so that someone could turn rollbar.js person tracking on or off, rather than have it be automatic.

**Comment by jondeandres** [comment]
Mmmm @and-megan, we even don't have that option in the gem itself, so I think it will be ok sending that information automatically. Using an external option would be the way to do it.

**Pull request merged**